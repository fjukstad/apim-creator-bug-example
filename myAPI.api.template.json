{
  "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
  "contentVersion": "1.0.0.0",
  "parameters": {
    "ApimServiceName": {
      "type": "string"
    },
    "serviceUrl": {
      "type": "object"
    }
  },
  "resources": [
    {
      "properties": {
        "type": "ht",
        "isCurrent": false,
        "subscriptionRequired": false,
        "displayName": "myAPI",
        "serviceUrl": "[parameters('serviceUrl').myAPI]",
        "path": "my-api",
        "protocols": [
          "https"
        ],
        "value": "openapi: \"3.0.0\"\n\ninfo:\n  title: myAPI\n  version: \"1.0\"\nservers:\n  - url: http://localhost:8000\npaths:\n  /:\n    get:\n      summary: myAPI\n      description: This is myAPI\n      operationId: hello.post_greeting\n      responses:\n        200:\n          description: the response\n          content:\n            text/plain:\n              schema:\n                type: string\n                example: \"hello from myAPI!\"\n",
        "format": "openapi",
        "apiType": "ht"
      },
      "name": "[concat(parameters('ApimServiceName'), '/myAPI')]",
      "type": "Microsoft.ApiManagement/service/apis",
      "apiVersion": "2019-01-01",
      "dependsOn": []
    }
  ]
}