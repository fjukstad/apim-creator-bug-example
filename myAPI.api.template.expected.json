{
  "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
  "contentVersion": "1.0.0.0",
  "parameters": {
    "ApimServiceName": {
      "type": "string"
    }
  },
  "resources": [
    {
      "properties": {
        "type": "ht",
        "isCurrent": false,
        "subscriptionRequired": false,
        "displayName": "myAPI",
        "path": "my-api",
        "protocols": [
          "https"
        ],
        "value": "openapi: \"3.0.0\"\ninfo:\n  title: myAPI\n  version: \"1.0\"\nservers:\n  - url: http://localhost:8000\npaths:\n  /:\n    get:\n      summary: myAPI\n      description: This is myAPI\n      operationId: hello.post_greeting\n      responses:\n        200:\n          description: the response\n          content:\n            text/plain:\n              schema:\n                type: string\n                example: \"hello from myAPI!\"\n",
        "format": "openapi",
        "apiType": "ht"
      },
      "name": "[concat(parameters('ApimServiceName'), '/myAPI')]",
      "type": "Microsoft.ApiManagement/service/apis",
      "apiVersion": "2019-01-01",
      "dependsOn": []
    },
    {
      "properties": {},
      "name": "[concat(parameters('ApimServiceName'), '/myProduct/myAPI')]",
      "type": "Microsoft.ApiManagement/service/products/apis",
      "apiVersion": "2019-01-01",
      "dependsOn": [
        "[resourceId('Microsoft.ApiManagement/service/apis', parameters('ApimServiceName'), 'myAPI')]"
      ]
    }
  ]
}